kv/kvserver: TestReplicateQueueSwapVotersWithNonVoters failed

kv/kvserver.TestReplicateQueueSwapVotersWithNonVoters [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/13095110?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/13095110?buildTab=artifacts#/) on release-22.2 @ [13fbbe977dd90e4b8c5530f27c5052ad3799ce06](https://github.com/cockroachdb/cockroach/commits/13fbbe977dd90e4b8c5530f27c5052ad3799ce06):


```
=== RUN   TestReplicateQueueSwapVotersWithNonVoters
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestReplicateQueueSwapVotersWithNonVoters55416342
    test_log_scope.go:79: use -show-logs to present logs inline
*
* INFO: Running test with the default test tenant. If you are only seeing a test case failure when this message appears, there may be a problem with your test case running within tenants.
*
    replicate_queue_test.go:1303: condition failed to evaluate within 45s: expected 4 non-voters; got [(n1,s1):5NON_VOTER (n4,s4):6NON_VOTER (n5,s5):7NON_VOTER]
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestReplicateQueueSwapVotersWithNonVoters55416342
--- FAIL: TestReplicateQueueSwapVotersWithNonVoters (96.95s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReplicateQueueSwapVotersWithNonVoters.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-34490

--- a/pkg/kv/kvserver/replicate_queue_test.go
+++ b/pkg/kv/kvserver/replicate_queue_test.go
@@ -1434,6 +1434,7 @@ func TestReplicateQueueSwapVotersWithNonVoters(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 	skip.UnderRace(t, "takes a long time or times out under race")
+	skip.UnderDeadlock(t, "takes a long time or times out under deadlock")
 
 	ctx := context.Background()
 	serverArgs := make(map[int]base.TestServerArgs)

